### PR TITLE
Updates and fixes

### LINKED ISSUE #14: Handle Ctrl+C better
```
Traceback (most recent call last):
  File "launch.py", line 4, in <module>
    main()
  File "E:\Development\Python\shanghai\shanghai\main.py", line 30, in main
    loop.run_until_complete(asyncio.wait(tasks))
  File "c:\python35\Lib\asyncio\base_events.py", line 330, in run_until_complete
    self.run_forever()
  File "c:\python35\Lib\asyncio\base_events.py", line 301, in run_forever
    self._run_once()
  File "c:\python35\Lib\asyncio\base_events.py", line 1162, in _run_once
    event_list = self._selector.select(timeout)
  File "c:\python35\Lib\selectors.py", line 314, in select
    r, w, _ = self._select(self._readers, self._writers, [], timeout)
  File "c:\python35\Lib\selectors.py", line 305, in _select
    r, w, x = select.select(r, w, w, timeout)
KeyboardInterrupt
<_WindowsSelectorEventLoop running=False closed=True debug=False>
{'task': <Task pending coro=<Network.start() running at E:\Development\Python\shanghai\shanghai\network.py:80> wait_for=
<_GatheringFuture pending cb=[Task._wakeup()]> cb=[_wait.<locals>._on_completion() at c:\python35\Lib\asyncio\tasks.py:4
00]>, 'message': 'Task was destroyed but it is pending!'}
Unhandled error in exception handler
context: {'task': <Task pending coro=<Network.start() running at E:\Development\Python\shanghai\shanghai\network.py:80>
wait_for=<_GatheringFuture pending cb=[Task._wakeup()]> cb=[_wait.<locals>._on_completion() at c:\python35\Lib\asyncio\t
asks.py:400]>, 'message': 'Task was destroyed but it is pending!'}
Traceback (most recent call last):
  File "c:\python35\Lib\asyncio\base_events.py", line 1066, in call_exception_handler
    self._exception_handler(self, context)
  File "E:\Development\Python\shanghai\shanghai\main.py", line 11, in exception_handler
    raise context['exception']
KeyError: 'exception'
<_WindowsSelectorEventLoop running=False closed=True debug=False>
{'task': <Task pending coro=<Client.run() running at E:\Development\Python\shanghai\shanghai\client.py:44> wait_for=<Fut
ure pending cb=[Task._wakeup()]> cb=[Network.runner_task_done(), gather.<locals>._done_callback(0)() at c:\python35\Lib\
asyncio\tasks.py:602]>, 'message': 'Task was destroyed but it is pending!'}
Unhandled error in exception handler
context: {'task': <Task pending coro=<Client.run() running at E:\Development\Python\shanghai\shanghai\client.py:44> wait
_for=<Future pending cb=[Task._wakeup()]> cb=[Network.runner_task_done(), gather.<locals>._done_callback(0)() at c:\pyth
on35\Lib\asyncio\tasks.py:602]>, 'message': 'Task was destroyed but it is pending!'}
Traceback (most recent call last):
  File "c:\python35\Lib\asyncio\base_events.py", line 1066, in call_exception_handler
    self._exception_handler(self, context)
  File "E:\Development\Python\shanghai\shanghai\main.py", line 11, in exception_handler
    raise context['exception']
KeyError: 'exception'
<_WindowsSelectorEventLoop running=False closed=True debug=False>
{'task': <Task pending coro=<Network.worker() running at E:\Development\Python\shanghai\shanghai\network.py:119> wait_fo
r=<Future pending cb=[Task._wakeup()]> cb=[Network.worker_task_done(), gather.<locals>._done_callback(1)() at c:\python3
5\Lib\asyncio\tasks.py:602]>, 'message': 'Task was destroyed but it is pending!'}
Unhandled error in exception handler
context: {'task': <Task pending coro=<Network.worker() running at E:\Development\Python\shanghai\shanghai\network.py:119
> wait_for=<Future pending cb=[Task._wakeup()]> cb=[Network.worker_task_done(), gather.<locals>._done_callback(1)() at c
:\python35\Lib\asyncio\tasks.py:602]>, 'message': 'Task was destroyed but it is pending!'}
Traceback (most recent call last):
  File "c:\python35\Lib\asyncio\base_events.py", line 1066, in call_exception_handler
    self._exception_handler(self, context)
  File "E:\Development\Python\shanghai\shanghai\main.py", line 11, in exception_handler
    raise context['exception']
KeyError: 'exception'
<_WindowsSelectorEventLoop running=False closed=True debug=False>
{'transport': <_SelectorSocketTransport fd=544>, 'exception': OSError(10038, 'Ein Vorgang bezog sich auf ein Objekt, das
 kein Socket ist', None, 10038, None), 'message': 'Fatal write error on socket transport', 'protocol': <asyncio.sslproto
.SSLProtocol object at 0x00000000033D7CC0>}
Unhandled error in exception handler
context: {'transport': <_SelectorSocketTransport fd=544>, 'exception': OSError(10038, 'Ein Vorgang bezog sich auf ein Ob
jekt, das kein Socket ist', None, 10038, None), 'message': 'Fatal write error on socket transport', 'protocol': <asyncio
.sslproto.SSLProtocol object at 0x00000000033D7CC0>}
Traceback (most recent call last):
  File "c:\python35\Lib\asyncio\base_events.py", line 1066, in call_exception_handler
    self._exception_handler(self, context)
  File "E:\Development\Python\shanghai\shanghai\main.py", line 11, in exception_handler
    raise context['exception']
  File "c:\python35\Lib\asyncio\selector_events.py", line 698, in write
    n = self._sock.send(data)
OSError: [WinError 10038] Ein Vorgang bezog sich auf ein Objekt, das kein Socket ist
<_WindowsSelectorEventLoop running=False closed=True debug=False>
{'transport': <_SelectorSocketTransport closing fd=544>, 'exception': RuntimeError('Event loop is closed',), 'message':
'Fatal error on SSL transport', 'protocol': <asyncio.sslproto.SSLProtocol object at 0x00000000033D7CC0>}
Unhandled error in exception handler
context: {'transport': <_SelectorSocketTransport closing fd=544>, 'exception': RuntimeError('Event loop is closed',), 'm
essage': 'Fatal error on SSL transport', 'protocol': <asyncio.sslproto.SSLProtocol object at 0x00000000033D7CC0>}
Traceback (most recent call last):
  File "c:\python35\Lib\asyncio\base_events.py", line 1066, in call_exception_handler
    self._exception_handler(self, context)
  File "E:\Development\Python\shanghai\shanghai\main.py", line 11, in exception_handler
    raise context['exception']
  File "c:\python35\Lib\asyncio\selector_events.py", line 698, in write
    n = self._sock.send(data)
OSError: [WinError 10038] Ein Vorgang bezog sich auf ein Objekt, das kein Socket ist

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\python35\Lib\asyncio\base_events.py", line 1066, in call_exception_handler
    self._exception_handler(self, context)
  File "E:\Development\Python\shanghai\shanghai\main.py", line 11, in exception_handler
    raise context['exception']
  File "c:\python35\Lib\asyncio\sslproto.py", line 624, in _process_write_backlog
    self._transport.write(chunk)
  File "c:\python35\Lib\asyncio\selector_events.py", line 702, in write
    self._fatal_error(exc, 'Fatal write error on socket transport')
  File "c:\python35\Lib\asyncio\selector_events.py", line 590, in _fatal_error
    self._force_close(exc)
  File "c:\python35\Lib\asyncio\selector_events.py", line 602, in _force_close
    self._loop.call_soon(self._call_connection_lost, exc)
  File "c:\python35\Lib\asyncio\base_events.py", line 453, in call_soon
    handle = self._call_soon(callback, args)
  File "c:\python35\Lib\asyncio\base_events.py", line 462, in _call_soon
    self._check_closed()
  File "c:\python35\Lib\asyncio\base_events.py", line 289, in _check_closed
    raise RuntimeError('Event loop is closed')
RuntimeError: Event loop is closed
```

### LINKED ISSUE #18: Handle 433 ERR_NICKNAMEINUSE

### PR DESCRIPTION
**Pull request by FichteFoll**
- Fixes #18 (https://github.com/chireiden/shanghai/commit/9ea8ad4eab2bc1b58096359748dc3a633a29e015)
- Fixes #14 (multiple commits)
- For more, see commit messages

### CODE REVIEW

File: shanghai/irc/options.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
+
+from . import Message
+
+
+# TODO evaluate against specs
+# http://www.irc.org/tech_docs/005.html
+# http://www.irc.org/tech_docs/draft-brocklesby-irc-isupport-03.txt
+class Options:
+    """A simple case insensitive mapping of 005 RPL_ISUPPORT reply."""
+    _fields = ('_options')
```
Comment (knitori) [inline shanghai/irc/options.py]: That's no tuple. Shouldn't it be one?

File: shanghai.example.yaml [hunk 2]
```diff
@@ -4,6 +4,7 @@ networks:
     # serves as base for each network
     # networks can overwrite any setting here.
     nick: Shanghai
+    user: Shanghai
```
Comment (knitori) [inline shanghai.example.yaml]: That's what `ident` was meant to be (I think it was meant to be used together with the `identd` service of a user in very old and very original versions of the internet). But `user` is what's being used in rfc, so I guess we should completely remove `ident`.

File: shanghai/network.py [hunk 3]
```diff
@@ -107,17 +112,22 @@ def start_register(self):
         assert event.name == "connected"
         # self.connection = event.value
         print(self.name, event)
-        running = True
+        stopped = False
 
         # start register process
         self.start_register()
-        while running:
+        while 1:
```
Comment (knitori) [inline shanghai/network.py]: `while True:` please :-)

**Comment by knitori** [comment]
CTRL+C still doesn't seem to fully work in some cases. I think it's because I set euirc to use SSL, but it couldn't really connect. And when it's trying to reconnect "after X seconds" it also fails to shut down properly by hitting ctrl+c once... Exceptions handling with asyncio gives me a bit of a headache. :-)

File: shanghai.example.yaml (hunk 2 above)
Comment (FichteFoll) [inline shanghai.example.yaml]: Afaik, user and ident are separate concepts. The "user" setting would be used for authentication (such as "Server PASS", or "SASL") while the ident doesn't have any meaning.

Hexchat requires "username" to be set in its config.

**Comment by FichteFoll** [comment]
`KeyboardInterrupt` is seemingly not raised when there is no task being run. This is especially true for while waiting for the reconnect but also for any other pause in the code, such as after connecting to the network, joining and then not receiving any message. If you press ctrl+c after that, the exception won't be raised until the server sends a message (e.g. a normal incoming message or PING).

File: shanghai/irc/options.py (hunk 1 above)
Comment (FichteFoll) [inline shanghai/irc/options.py]: It should be indeed.

**Comment by FichteFoll** [comment]
I'll fix the non-tuple error and rebase this onto master. There is no need to create a merge commit if the commit history if there were no other changes.

File: shanghai.example.yaml (hunk 2 above)
Comment (knitori) [inline shanghai.example.yaml]: No well.. the `user` is currently used for the [`USER`](https://tools.ietf.org/html/rfc2812#section-3.1.3) command upon connecting. In rfc1459 the username, specified for the `USER` command, had a [different meaning](https://tools.ietf.org/html/rfc1459#section-4.1.3), that's been removed in rfc2812 as it seems. There is also a section about [ident lookups](https://tools.ietf.org/html/rfc1459#section-8.11.2) in the old rfc. But this is pretty much obsolete nowadays. So I think it's fine to remove the ident argument.

Comment (FichteFoll) [inline shanghai.example.yaml]: I'll let you make the changes to this since you apparently had some local changes regarding it.

**Pull request merged**